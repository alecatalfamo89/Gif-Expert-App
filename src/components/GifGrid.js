import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifts';

import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {


const {data:images,loading} = useFetchGifs(category);

// , { useState,useEffect }

  
  return (
    <>
    <h3 className='text animate__fadeIn'>{category}</h3>  
      {loading && <p className='wep animate__flash'>Loading...</p>}
    <div className='card-grid'>       
        
          {
            images.map(img => (
              
            <GifGridItem
             key={img.id}
             {...img}             
            />
            
            ))
            }
           

    </div>
    </>
  )  
}
