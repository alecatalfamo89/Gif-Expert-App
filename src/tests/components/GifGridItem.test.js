import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';


describe('pruebas en GifGridItem', () => { 
    const title = 'Un Titulo';
    const url = 'https/localhost.jpg';
    test('debe mostrar el componente correctamente', ()=>{
        const wrapper = shallow( <GifGridItem title={title} url={url} /> )
        expect(wrapper).toMatchSnapshot();
    })
 })