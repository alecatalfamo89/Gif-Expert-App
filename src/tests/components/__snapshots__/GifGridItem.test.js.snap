// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="Un Titulo"
    src="https/localhost.jpg"
  />
  <p>
     
    Un Titulo
  </p>
</div>
`;
